pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/KafkaConfluentSchemaRegistryAvroMessageDecoderRealtimeClusterIntegrationTest.java

```diff
@@ -0,0 +1,313 @@
+
+  @Override
+  protected int getNumKafkaBrokers() {
+    return 1;
+  }
+
+  @Override
+  protected void startKafka() {
+    super.startKafka();
+    startSchemaRegistry();
+  }
+
+  @Override
+  protected void stopKafka() {
+    stopSchemaRegistry();
+    super.stopKafka();
+  }
+
+  private void startSchemaRegistry() {
+    if (_schemaRegistry == null) {
+      _schemaRegistry = SchemaRegistryStarter.startLocalInstance(SchemaRegistryStarter.DEFAULT_PORT);
+    }
+  }
+
+  private void stopSchemaRegistry() {
+    try {
+      if (_schemaRegistry != null) {
+        _schemaRegistry.stop();
+        _schemaRegistry = null;
+      }
+    } catch (Exception e) {
+      // Swallow exceptions
+    }
+  }
+
+  @Override
+  protected void pushAvroIntoKafka(List<File> avroFiles)
+      throws Exception {
+    Properties avroProducerProps = new Properties();
+    avroProducerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:" + getKafkaPort());
+    avroProducerProps.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, _schemaRegistry.getUrl());
+    avroProducerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.ByteArraySerializer");
+    avroProducerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+        "io.confluent.kafka.serializers.KafkaAvroSerializer");
+    Producer<byte[], GenericRecord> avroProducer = new KafkaProducer<>(avroProducerProps);
+
+    Properties nonAvroProducerProps = new Properties();
+    nonAvroProducerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:" + getKafkaPort());
+    nonAvroProducerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.ByteArraySerializer");
+    nonAvroProducerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+        "org.apache.kafka.common.serialization.ByteArraySerializer");
+    Producer<byte[], byte[]> nonAvroProducer = new KafkaProducer<>(nonAvroProducerProps);
+
+    if (injectTombstones()) {
+      // publish lots of tombstones to livelock the consumer if it can't handle this properly
+      for (int i = 0; i < 1000; i++) {
+        // publish a tombstone first
+        nonAvroProducer.send(
+            new ProducerRecord<>(getKafkaTopic(), Longs.toByteArray(System.currentTimeMillis()), null));
+      }
+    }
+    for (File avroFile : avroFiles) {
+      try (DataFileStream<GenericRecord> reader = AvroUtils.getAvroReader(avroFile)) {
+        for (GenericRecord genericRecord : reader) {
+          byte[] keyBytes = (getPartitionColumn() == null) ? Longs.toByteArray(System.currentTimeMillis())
+              : (genericRecord.get(getPartitionColumn())).toString().getBytes();
+          // Ignore getKafkaMessageHeader()
+          nonAvroProducer.send(new ProducerRecord<>(getKafkaTopic(), keyBytes, "Rubbish".getBytes(UTF_8)));
+          avroProducer.send(new ProducerRecord<>(getKafkaTopic(), keyBytes, genericRecord));
```

Comment:
Hi @ddcprg can you help us understand if we need to produce one non-avro rubbish per avro producer row?
we are seeing a lot of rubbish in the logs and we were trying to determine whether we should slim down the number of rubbish rows published in this test case. 

see: #9516

Reply:
Hey @walterddr not at all, we only need to produce a few messages to make sure they are skipped. Let me know if you like me to raise a PR

Reply:
No worries. We will limit the rubbish rows produced thank you for confirming!